test/e2e/framework/util.go

```diff
@@ -684,14 +641,14 @@ func getControlPlaneAddresses(ctx context.Context, c clientset.Interface) ([]str
 	var externalIPs, internalIPs, hostnames []string
 
 	// Populate the internal IPs.
-	eps, err := c.CoreV1().Endpoints(metav1.NamespaceDefault).Get(ctx, "kubernetes", metav1.GetOptions{})
+	slices, err := c.DiscoveryV1().EndpointSlices(metav1.NamespaceDefault).List(ctx, metav1.ListOptions{LabelSelector: discoveryv1.LabelServiceName + "=kubernetes"})
 	if err != nil {
 		Failf("Failed to get kubernetes endpoints: %v", err)
 	}
-	for _, subset := range eps.Subsets {
-		for _, address := range subset.Addresses {
-			if address.IP != "" {
-				internalIPs = append(internalIPs, address.IP)
+	for _, slice := range slices.Items {
+		for _, endpoint := range slice.Endpoints {
```

Comment:
ditto, endpoints on endpointslices has the ready condition, to be the same logic as we had with Endpoints we need to check that